lib/Server.js

```diff
@@ -58,6 +59,8 @@ class Server {
     this.compiler = compiler;
     this.options = options;
 
+    setPublicPath(compiler, options);
```

Comment:
Let's move this in `updateCompiler`

Reply:
@evilebottnawi this is not altering the compiler at all though, only using the compiler's option data to set `devServer.publicPath`. But if you still think it should be there I have no strong opinion.